documentation/sdn/ovn_bgp_evpn.md

```diff
@@ -0,0 +1,125 @@
+# Using OVN BGP Agent with Evpn driver in Openstack Kolla
+
+
+This article covers the basics of BGP and EVPN and their applications. It focuses on the advantages of using OVN with the OVN-BGP agent compared to plain OVN.
+
+## BGP
+Border Gateway Protocol (BGP) is the protocol used to exchange routing information between different networks. It's a key component that helps the Internet function by determining the best paths for data to travel from one point to another.
+
+## OVN BGP AGENT
+
+The OVN BGP Agent is a component designed to integrate the Open Virtual Network (OVN) with the Border Gateway Protocol (BGP) for routing purposes. OVN is a network virtualization system that provides software-defined networking (SDN) capabilities for virtualized environments, such as OpenStack. BGP, on the other hand, is the protocol used to exchange routing information between different networks (as explained in the previous message).
+
+### Advantages of OVN BGP vs pure OVN networking
+- **Direct external network reachability / connectivity** - External access to VMs and  connectivity of VMs to external network  is simplified because the BGP agent advertises their internal IP addresses without needing NAT, Floating IP or external gateway. By using BGP, which is a standard protocol in networking, the OVN BGP Agent integrates OVN virtual networks seamlessly with existing physical routers and external network infrastructures. This allows OVN environments to be easily connected to external data centers, service provider networks, or the Internet. Without OVN BGP Agent, connecting requires configuring gateway routers or L3 devices to handle the routing between the virtual and physical networks.
```

Comment:
I think this lacks some realworld use case examples. Assuming we are talking about IPv4, in most use cases the customers want/should use RFC1918 / private addressing between VMs (keep in mind that IPv4 addresses are rare and expensive, we charge ~3 EUR per month for one!). Exposing them externally would not mean external connectivity without introducing NAT somewhere else.

This is of course different for IPv6 - which is not using NAT either way (nor with or without bgp agent)

Also, this seems to cover the L3 mode "afi IPv4/6 unicast", but below we are talking about the L2 "EVPN" mode. When we experimented with the OVN BGP Agent I was mostly interested in the "expose Floating IPs and IPv6 subnets to switches (L3)" part, but EVPN is a totally different use case, which might make a lot of sense for interconnectivity. Maybe I'm missing something here, but for me it's not clear.